### ISSUE TITLE
number_format filter and negative values

### ISSUE THREAD

**Issue by DarkBee** [twigphp/Twig#2470]
Using the `number_format` filter along with a negative number shows some weird output.

```
{{ 9800.333|number_format(2) }} {# output : 9,800.33 #}
{{ -9800.333|number_format(2) }} {# output : -9 #}
```

Running this in PHP,

```
<?php
       var_dump(number_format(9800.333, 2)); // output 9800.33
	var_dump(number_format(-9800.333, 2)); // output -9800.33
	var_dump(number_format("9800.333", 2)); // output 9800.33
	var_dump(number_format("-9800.333", 2)); // output -9800.33
```

I know this can be fixed by adding parentheses around the variable in twig but this still feels off.
Don't understand how the output can become 9 either

**Comment by jvasseur** [comment]
This is due to twig operator precedence (and some PHP strangeness): `-9800.333|number_format(2)` is interpreted as `-number_format(9800.333, 2)` that get resolver by php as `-"9,800.33"` php ignoring everything after the `,` when converting the string back to an int we get `-9`.

And we can't change the operator precedence without breaking BC so I don't think there is a solution for this problem.

**Comment by DarkBee** [comment]
I see, perhaps this issue should added as a warning in the [official documents](https://twig.sensiolabs.org/doc/2.x/filters/number_format.html) then

**Comment by stof** [comment]
the solution is to use braces, as braces win for the precedence (like in any other language): `(-9800.333)|number_format(2)`

**Comment by DarkBee** [comment]
That I already pointed out in my initial post, it just does not feel natural for users to place parentheses around a negative number. 

Your comment "like in any other language" does not make any sense though..
In PHP you can do `number_format(-9800.333, 2)` without any problem regarding precedence

**Comment by stof** [comment]
@DarkBee but in PHP, you don't involve an operator for the formatting. You involve a function call. So there is no precedence here.

Changing the precedence of the filter operator would be a huge BC break.

**Comment by DarkBee** [comment]
I'm not suggesting to change the precedence of the filter operator but to add a warning on the documentation about this at the number_format section

**Comment by fabpot** [comment]
@DarkBee Can you work on a doc change that would add a warning about this?

**Comment by DarkBee** [comment]
@fabpot I've posted a pull [request ](https://github.com/twigphp/Twig/pull/2473/commits) on the matter

**Comment by fabpot** [comment]
Fixed in #4543
